static/js/containers/VideoDetailPage.js

```diff
@@ -147,33 +161,58 @@ class VideoDetailPage extends React.Component {
           Uploaded {formattedCreation}
         </span>
         <span className="actions">
-          <Button className="edit mdc-button--raised" onClick={this.openDialog} disabled={!editable}>
+          <Button className="edit mdc-button--raised" onClick={this.openEditDialog} disabled={!editable}>
             <span className="material-icons">edit</span> Edit
           </Button>
-          <Button className="share mdc-button--raised">
+          <Button className="share mdc-button--raised"  onClick={this.openshareDialog}>
             <span className="material-icons">share</span> Share
           </Button>
         </span>
       </div>
       <Footer />
       <Dialog
-        open={videoDetailUi.dialog.visible}
+        open={videoDetailUi.editDialog.visible}
         onAccept={this.submitForm}
-        onCancel={this.clearDialog}
+        onCancel={this.clearEditDialog}
         title="Edit video details"
+        id="editDialog"
       >
         <div className="mdc-form-field mdc-form-field--align-end">
           <Textfield
             label="New title:"
             id="video-title"
             onChange={this.setTitle}
-            value={videoDetailUi.dialog.title}
+            value={videoDetailUi.editDialog.title}
           />
           <Textfield
             label="Description:"
             id="video-description"
             onChange={this.setDescription}
-            value={videoDetailUi.dialog.description}
+            value={videoDetailUi.editDialog.description}
+          />
+        </div>
+      </Dialog>
+      <Dialog
+        open={videoDetailUi.shareDialog.visible}
+        title="Share this Video"
+        onCancel={this.clearshareDialog}
+        cancelText="Close"
+        noSubmit={true}
+        id="shareDialog"
+      >
+        <div className="mdc-form-field mdc-form-field--align-end">
+          <Textfield
+            readOnly
+            label="Video URL:"
+            id="video-url"
+            value={embedUrl}
```

Comment:
can we name this variable something like `videoDetailURL`? `embedUrl` doesn't quite seem like an accurate name

Reply:
How about videoShareURL?

Reply:
that sounds good to me!